### ISSUE TITLE
boxblur maxes out memory

### ISSUE THREAD

**Issue by beauterre** [licson0729/CanvasEffects#4]
I tried the boxblur, but it keeps saying: not enough memory to open the page.
I assume you are trying to do a blur by some sort of convolution filter?

Quick TIP:
You could do a very simple blur, by just drawing (a copy of) this image over itself, but shifted one pixel in x+1 and with a globalAlpha=1/2;
Then you do it again with x-1 and globalAlpha=1/3;
Again with y-1 and globalAlpha=1/4
And lastly again with y_1 and globalAlpha=1/5

This (globalAlpha=1/nrofoperations)  simulates an accumulation buffer, but is REALLY fast and consumes very little memory, because it uses the build in functions. Al you need is a clean copy of your image to blit.

With a bit of fiddling a gaussian blur is also possible and with applycation of the difference, screen and multiply operators some awesome filtering is possible :)
Hope this helps.

**Comment by licson** [comment]
Yes, the blur effect is currently implemented as a convolution filter which yields best visual results. Though I haven't come across a memory issue when testing the effect in both Firefox and Chrome browsers. 

I am thinking of a more efficient way to implement various types of blur in the future.
